app/views/collection/show.html.slim

```diff
@@ -116,14 +116,19 @@
       =label_tag 'search', t('.search_by_title_1'), class: 'hidden'
 
     -if @collection.active? && @collection.text_entry?
+      ruby:
+        pages_need_correction_count = Page.where(work_id: @collection.work_ids, status: [nil, Page::STATUS_INCOMPLETE]).count
+        pages_need_review_count = Page.where(work_id: @collection.work_ids, status: Page::STATUS_NEEDS_REVIEW).count
```

Comment:
Although this accomplishes the task, I think it might be best not to do this?

Question would be do we want this feature even at the cost of performance?

That being said I'm thinking of more optimal way to do this, let me know if you have ideas as well

Reply:
We roll up this kind of data in the `work_statistics` table, which is a child of `works` and gets re-calculated on any page status change.  Rewriting this do something like this (pseudocode) would be worth the performance trade-off, even if the we have to add the computations (and migrations) to `WorkStatistic`.
`@collection.joins([:works, :work_statistics]).where("work_statistic.needs_review_pages > 0").present?`

Reply:
@benwbrum I see no issue for pages_need_review_count. However for pages_need_correction I think we need other than incomplete_pages_count correct?

I think no change is needed here after all. We can perhaps test the performance first before deciding if this needs further optimization?

Reply:
I think that we can use `work_statistics` in this case by checking whether the sum of `work_statistics.total_pages` - (`work_statistics.completed_pages` + `work_statistics.needs_review_pages`) is non-zero.  It is not clear to me which approach would work better, but doing a couple of tests on a reasonably large collection and observing the rack miniprofiler would probably let us decide.

Reply:
I think doing the current one is much more efficient/accurate. The rollup counts does not seem to match the actual records on some particular works. I outlined my reasoning in slack, let me know if you have any other questions @benwbrum